pkg/daemon/update.go

```diff
@@ -1716,6 +1716,16 @@ func (dn *Daemon) updateSSHKeys(newUsers []ign3types.PasswdUser) error {
 		return nil
 	}
 
+	_, err := user.Lookup(constants.CoreUserName)
+	switch err.(type) {
```

Comment:
Instead of using a switch for verifying error type, use `errors.Is()` instead:

```golang
_, err := user.Lookup(constants.CoreUserName)
if err == nil {
  return nil
}

if errors.Is(err, user.UnknownUserError) {
  // Log
}

return fmt.Errorf("failed to check if user core exists: %w", err)
```

Reply:
one correction, `if err == nil {` , we should continue the further steps to write the sshKey instead of returning nil

Reply:
Changed to use `errors.As` in the switch